libs/sdk-core/src/breez_services.rs

```diff
@@ -1368,6 +1370,36 @@ impl BreezServices {
             },
         })
     }
+
+    pub async fn register_notifications(
```

Comment:
Isn't it better to get the url directly from the caller? Also perhaps we want to rename this function to `register_webhook` ?
The rational is:
1. This should also be used as a general web hook, not only for mobile
2. The vendor that is running the NDS may decide on a different url format than we introduce here.

Reply:
> This should also be used as a general web hook

Then we might need an extra arg for the type of event that triggers this webhook, like so:

```rust
pub enum WebhookEvent {
    PaymentReceived
}

register_webhook(&self, callback_url: String, event: WebhookEvent) -> SdkResult<()>
```

That might also be needed because we cannot call the same `callback_url` for every notification type, because they have different payloads so the caller needs to choose an appropriate `callback_url`.

Would you agree?

Reply:
Alternatively the event type could be included in the call to the webhook url:

```
<base_url>?token=<token>&type=<type>
```

Reply:
Good point. Both are valid options.

I've added it as an open question in the umbrella issue #574. IMO for now we can leave it as it is, we only have one notification type.